hoprd/rest-api/src/channels.rs

```diff
@@ -290,11 +290,18 @@ pub(super) async fn open_channel(
 ) -> impl IntoResponse {
     let hopr = state.hopr.clone();
 
+    let destination = open_req.clone().destination.fulfill(hopr.peer_resolver()).await;
```

Comment:
If you made your `fulfil` method be `fn fulfill(self, resolver: ...) -> PeerOrAddress` instead of `fn fulfill(&mut self, resolver: ...)`, 
you'd avoid doing any calls to `clone()` every time you need - it would be clone free.

You can now see you've made the `fulfill` method less ergonomic in the most common use case.

Even worse, you're now cloning the entire `args` structure, whereas cloning of `args.destination` would've been sufficient.